types/googlemaps/reference/map.d.ts

```diff
@@ -612,6 +612,13 @@ declare namespace google.maps {
          * occurred.
          */
         latLng: LatLng;
+        /**
+         * The corresponding native DOM event.  Developres hsould not rely on target,
+         * currentTarget, relatedTarget and path properties being defined and 
+         * consistent.  Developres should not also rely on DOM structure of hte internal
+         * implementation of Maps API.
+         */
+        domEvent: lib.dom.MouseEvent  | lib.dom.TouchEvent | lib.dom.PointerEvent | lib.dom.Event;                                                  
```

Comment:
Also, I'm not sure whether `lib.dom` needs to be referenced here? These events should be in the global scope already afaik.

```suggestion
        domEvent: MouseEvent  | TouchEvent | PointerEvent | Event;                                                  
```

Reply:
Ya wow. Those typos are terrible. I should have copy paste dit but I didn't think I'd mess so much up.

`lib.dom` is required (at least locally) otherwise `MouseEvent` errors out as its the same name as the Google type and typescript complains.

Unsure how to deal with that without fully qualifying the Dom type (sorry, any pointers?).

Reply:
This week I will try to update the type names to avoid collisions.

Reply:
I updated all the `MouseEvent` references to `MapMouseEvent` where applicable.  Some of the extended APIs still use `MouseEvent` but those are all namespaced with `data`, `visualization`, etc.

This is definitely a breaking change, but I'm not entirely sure how to update the version.  The current version is `3.40`, while the latest Google Maps version is `3.43` which would not constitute a major version change.  The next logical would be `4.0` but this would no longer match with the Google Maps version which seems to be the case now.

Anyhow, hopefully this helps.